examples/fedopt_example/client.py

```diff
@@ -1,176 +1,93 @@
-            optimizer.zero_grad()
-            out = model(data, (h0, c0))
-            loss = criterion(out, labels)
-            loss.backward()
-            optimizer.step()
-
-            running_loss += loss.item()
-            _, predicted = torch.max(out.data, 1)
-
-            # Report some batch loss statistics every so often to track decrease
-            if batch_index % 100 == 0:
-                log(INFO, f"Batch Index {batch_index} of {n_batches}, Batch loss: {loss.item()}")
-            epoch_metrics.update_performance(predicted, labels)
-
-        log_str = epoch_metrics.summarize()
-        # Local client logging of epoch results.
-        log(
-            INFO,
-            f"Epoch: {epoch}, Client Training Loss: {running_loss/n_batches}\nClient Training Metrics:{log_str}",
-        )
-    return epoch_metrics.results
-
-
-def validate(
-    model: nn.Module,
-    validation_loader: DataLoader,
-    label_encoder: LabelEncoder,
-    device: torch.device = torch.device("cpu"),
-) -> Tuple[float, Metrics]:
-    """Validate the network on the entire validation set."""
-    model.eval()
-    criterion = torch.nn.CrossEntropyLoss()
-    loss = 0.0
-
-    assert validation_loader.batch_size is not None
-    assert isinstance(model, LSTM)
-    h0, c0 = model.init_hidden(validation_loader.batch_size)
-    h0 = h0.to(device)
-    c0 = c0.to(device)
-
-    epoch_metrics = ClientMetrics(label_encoder)
-
-    model.eval()
-    with torch.no_grad():
-        n_batches = len(validation_loader)
-        for data, labels in validation_loader:
-            data, labels = data.to(device), labels.to(device)
-            out = model(data, (h0, c0))
-            loss += criterion(out, labels).item()
-            _, predicted = torch.max(out.data, 1)
-            epoch_metrics.update_performance(predicted, labels)
-
-    log_str = epoch_metrics.summarize()
-    # Local client logging.
-    log(
-        INFO,
-        f"Client Validation Loss: {loss/n_batches}\nClient Validation Metrics:{log_str}",
-    )
-    return loss / n_batches, epoch_metrics.results
-
-
-class NewsClassifier(NumpyFlClient):
-    def __init__(self, data_path: Path, device: torch.device) -> None:
-        super().__init__(data_path, device)
-        self.parameter_exchanger = FullParameterExchanger()
-
-    def setup_client(self, config: Config) -> None:
-        super().setup_client(config)
+from fl4health.checkpointing.checkpointer import TorchCheckpointer
+from fl4health.clients.basic_client import BasicClient
+from fl4health.utils.losses import LossMeter, LossMeterType
+from fl4health.utils.metrics import MetricMeterManager
+
+
+class NewsClassifierClient(BasicClient):
+    def __init__(
+        self,
+        data_path: Path,
+        device: torch.device,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        checkpointer: Optional[TorchCheckpointer] = None,
+    ) -> None:
+        super(BasicClient, self).__init__(data_path, device)
```

Comment:
Note: This is a slight hack that likely won't be necessary with additional refactors to the metrics managers class. Since I'm not explicitly defining a set of metric for my MetricMeter class, I am skipping over the `__init__` function of the `BasicClient` class and directly going to the NumpyFl parent `__init__`

Reply:
Just for future reference, for #69 I slightly refactored how we handle accumulating and computing metrics, so I decided to base off this branch and update it so we call the BasicClient constructor, and save us from having to redefine a lot of attributes.